test/unit/native_deps/native_deps_test.bzl

```diff
@@ -140,19 +148,27 @@ def _cdylib_has_native_dep_and_alwayslink_test_impl(ctx):
             "-lstatic=native_dep{}".format(pic_suffix),
             "link-arg=-Wl,-force_load,bazel-out/{}-{}/bin/{}test/unit/native_deps/libalwayslink{}.lo".format(darwin_component, compilation_mode, workspace_prefix, pic_suffix),
         ]
+        asserts.equals(env, want, linker_args)
     elif ctx.target_platform_has_constraint(ctx.attr._windows_constraint[platform_common.ConstraintValueInfo]):
-        want = [
+        want_msvc = [
             "-lstatic=native_dep",
             "link-arg=/WHOLEARCHIVE:bazel-out/x64_windows-{}/bin/{}test/unit/native_deps/alwayslink.lo.lib".format(compilation_mode, workspace_prefix),
         ]
+        want_gnu = [
+            "-lstatic=native_dep",
+            "link-arg=-Wl,--whole-archive",
+            "link-arg=bazel-out/x64_windows-{}/bin/{}test/unit/native_deps/alwayslink.lo.lib".format(compilation_mode, workspace_prefix),
+            "link-arg=-Wl,--no-whole-archive",
+        ]
+        asserts.true(env, want_msvc == linker_args or want_gnu == linker_args)
```

Comment:
Is there a way you can know explicitly which one to test against? Doesn't fundamentally the `rust_toolchain.exec_triple` control these flags? So even though there isn't a given way to represent `msvc` vs `gnu` with the `@platforms` repo, you could still check within this function.

Reply:
`rust_toolchain.target_triple` controls them.  Though really it's the linker flavor that controls them, but as far as I'm aware there's not a way to successfully mix those, like saying `ld64.lld` + `x86_64-pc-windows-msvc` so it's not an important distinction.

Is there a way to get access to the toolchain instance inside of the test here?  That would be a vastly better solution 🙂